internal/subscription/core.go

```diff
@@ -343,54 +342,36 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
 
 	logger.Ctx(ctx).Infow("migration: found subscriptions to migrate", "count", len(subscriptionsToUpdate))
 
-	topicNames := make([]string, 0)
-	for _, model := range subscriptionsToUpdate {
+	updatedSubCount := 0
+	for _, sub := range subscriptionsToUpdate {
```

Comment:
While acting on all objects of an entity always batch the operation. This seems harmless now but down the line a migration can run for multiple hours and consume a large memory footprint (Need not change as part of this PR)